modules/imgproc/src/color_lab.cpp

```diff
@@ -1585,6 +1585,97 @@ struct RGB2Lab_b
 
         i = 0;
 
+#if CV_NEON
+
+        // Define some scalar constants which we will make use of later
+        const int one_lsh_by_lab_shift_sub_one = (1 << ((lab_shift)-1));
+        const int one_lsh_by_lab_shift2_sub_one = (1 << ((lab_shift2)-1));
+        const int one_shift_lab_mul_128 = 128*(1 << lab_shift2);
+
+        // On each loop, we load nlanes of RGB/A v_uint8s and store nlanes of
+        // Lab v_uint8s
+        for(; i <= n - v_uint8::nlanes; i += v_uint8::nlanes,
+                src += scn*v_uint8::nlanes, dst += 3*v_uint8::nlanes )
+        {
+            // Load 4 batches of 4 src
+            // =======================
+            v_uint8 vRi, vGi, vBi;
+            if(scn == 4)
+            {
+                v_uint8 vAi;
+                v_load_deinterleave(src, vRi, vGi, vBi, vAi);
+            }
+            else // scn == 3
+            {
+                v_load_deinterleave(src, vRi, vGi, vBi);
+            }
+
+#define RGB2LAB_BATCH(n) \
```

Comment:
May I suggest to use a static (inline?) function here instead of a macro? Should make the code easier to read. It will likely be inlined so the final code will perform as the one produced with the macro.

You should also guard it with `CV_NEON` to prevent it being built when not used.

Reply:
Done, although it was neater to do in a member function of RGB2Lab_b